src/components/coin.ts

```diff
@@ -90,6 +129,31 @@ export const adjustCoinBalanceByUserId = async (userId: string, amount: number):
     amount,
     amount
   );
+  const newBalance = await getCoinBalanceByUserId(userId);
+  await createCoinLedgerEntry(userId, newBalance - oldBalance, event, reason, adminId);
```

Comment:
is it possible to use `amount` instead of calculating the diff using newBalance and oldBalance here?

Reply:
I don't think so. As can be seen in the SQL command, the max is taken with 0, to take into account a negative adjustment with too large of a magnitude.

Reply:
If we need to get the old amount anyways, I think we can just calculate the new balance and re-use the update function to simplify logic. What do you think?

Reply:
Won't calculating the new balance need the old balance? In that case, won't it be simpler to just call `createCoinLedgerEntry` instead of `updateCoinBalanceByUserId`?

Reply:
Hmm I see what you mean, I was trying to find a way to narrow the write down to one code path but I guess we need to get old balance either way.

We can keep the create ledger entry logic here then, but we can pre-compute the amount diff in memory? So we can avoid an extra DB call, and it makes the SQL simpler too. Also don't forget to add TODOs to wrap these in a transaction.